commons/src/test/java/org/openmrs/eip/DeleteDataTestExecutionListener.java

```diff
@@ -3,90 +3,136 @@
  */
 package org.openmrs.eip;
 
-import java.sql.Connection;
-import java.sql.DatabaseMetaData;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.sql.Statement;
-import java.util.ArrayList;
-import java.util.List;
-
-import javax.sql.DataSource;
-
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.context.ApplicationContext;
 import org.springframework.test.context.TestContext;
 import org.springframework.test.context.support.AbstractTestExecutionListener;
 
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
- * Custom TestExecutionListener that resets all tables in the database by deleting all rows in them.
- * Typically, this listener should be configured to run after every test method has executed
+ * Custom TestExecutionListener that resets all tables in the database by deleting all rows in them. Typically, this listener
+ * should be configured to run after every test method has executed
  */
 public class DeleteDataTestExecutionListener extends AbstractTestExecutionListener {
 	
 	protected final Logger log = LoggerFactory.getLogger(getClass());
 	
 	private static final String DELETE = "DELETE FROM ";
 	
-	private static final String DISABLE_KEYS = "SET FOREIGN_KEY_CHECKS=0";
-	
-	private static final String ENABLE_KEYS = "SET FOREIGN_KEY_CHECKS=1";
-	
 	/**
 	 * @see AbstractTestExecutionListener#afterTestMethod(TestContext)
 	 */
 	@Override
 	public void afterTestMethod(TestContext testContext) throws Exception {
-		ApplicationContext ctx = testContext.getApplicationContext();
-		DataSource dataSource = ctx.getBean(Constants.MGT_DATASOURCE_NAME, DataSource.class);
+		var applicationContext = testContext.getApplicationContext();
+		var dataSource = applicationContext.getBean(Constants.MGT_DATASOURCE_NAME, DataSource.class);
 		
 		log.debug("Deleting all data from management DB tables...");
-		try (Connection c = dataSource.getConnection()) {
-			deleteAllData(c, "TEST");
+		try (Connection connection = dataSource.getConnection()) {
+			deleteAllData(connection, "TEST", "h2");
 		}
 		
-		dataSource = ctx.getBean(Constants.OPENMRS_DATASOURCE_NAME, DataSource.class);
+		dataSource = applicationContext.getBean(Constants.OPENMRS_DATASOURCE_NAME, DataSource.class);
 		
 		log.debug("Deleting all data from OpenMRS DB tables...");
 		
-		try (Connection c = dataSource.getConnection()) {
-			deleteAllData(c, "openmrs");
+		try (Connection connection = dataSource.getConnection()) {
+			deleteAllData(connection, "openmrs", "mysql");
+		}
+	}
+	
+	/**
+	 * Returns the SQL statement to disable foreign key checks for the specified DBMS
+	 * @param dbms the DBMS
+	 * @return the SQL statement to disable foreign key checks for the specified DBMS
+	 */
+	private static String DISABLE_FOREIGN_KEY_CHECKS(String dbms) {
```

Comment:
I find this naming an anti-pattern against the usual java method naming conventions.